Allow editing words from admin panel

AlejandroRG31 opened the pull request:
Added new ways to search words and entries in the admin panel to make easier edit the words

Added links in the frontend in the search result to the word/entry in the admin panel to edit the data by selecting them from the frontend

AlejandroRG31 committed:
Added new search bars and filters to admin panel to make easier edit the data

AlejandroRG31 committed:
Added django.contrib.admin and messages to installed apps

AlejandroRG31 committed:
Added id in entry serializer and removed admin url in the backend

AlejandroRG31 committed:
Merge branch 'master' into feature/edit-entries-admin-panel

AlejandroRG31 committed:
Fixed some flake8 errors

AlejandroRG31 committed:
Added a list filter in entry to filter by region

AlejandroRG31 committed:
Merge branch 'master' into feature/edit-entries-admin-panel

dunetna commented on linguatec_lexicon/admin.py:
```diff
@@ -2,24 +2,48 @@
-    list_display = ('translation', 'word',)
+    list_display = ('word', 'translation', 'variation')
+    search_fields = ('word__term',)
+    list_filter = ('word__lexicon', 'word__lexicon__src_language',
+                   'word__lexicon__dst_language', 'variation__region__name')
+
```
* The filter 'word__lexicon' does not appear in filter list.
* The filter 'variation__region__name' appears as "By name" filter, could this text be more specific?

AlejandroRG31 replied:
About the filter 'variation__region__name', I have added a new function which allows to change the name in a easy way, so now the title of the list is by Region instead of by name, thank you for the advice.

About 'word__lexicon' not appearing it is strange because it appears in my project and in my admin panel so i dont know what can be the problem.

![image](https://user-images.githubusercontent.com/64277128/100552093-408db500-3285-11eb-811d-e3874400be7d.png)

dunetna replied:
I only had one lexicon, adding another one, the filter appears. But it's quite weird because I have no regions and region filter is visible.

AlejandroRG31 replied:
I think this is happening because of how the method filter_list behaves when is used with a foreign key variable that can be null, and in our case the entries which do not have a specific variation their variation field take a null/none value, so I have been searching for function that could solve this problem removing the region list when there are no values in the list but I could not have found one.

About the fact that the filter lexicon does not appear unless there are more than one lexicon added in the database, I don't know why it behaves like that, maybe because having only one lexicon in the database which can't be a null/none variable so having it makes it redundant. Although, if this was the case neither src_language nor dst_language should appear.

What do you think? Do you know a method that could solve this?

slamora replied:
I think that this behaviour is the django defaults. So, no need to pay attention on changing that.

dunetna commented on linguatec_lexicon/admin.py:
```diff
@@ -2,24 +2,48 @@
 class ExampleAdmin(admin.ModelAdmin):
     list_display = ('phrase', 'entry',)
+    search_fields = ('entry__word__term',)
+    list_filter = ('entry__word__lexicon', 'entry__word__lexicon__src_language', 'entry__word__lexicon__dst_language',)
```
* The filter 'entry__word__lexicon' does not appear in filter list.

AlejandroRG31 replied:
About it not appearing, it is strange because it appears in my project and in my admin panel so i dont know what can be the problem.

![image](https://user-images.githubusercontent.com/64277128/100552054-02909100-3285-11eb-9c61-239046cb4f8e.png)

dunetna replied:
Idem as previous comment.

AlejandroRG31 committed:
Added a new function in admin.py which allows choosing the name you want to be displayed in the filter list as title.

AlejandroRG31 committed:
Moved admin urls from frontend to backend in order to have it in the project

AlejandroRG31 committed:
Included in the backend API the url to access the entry in the admin panel

Also added in the serializers.py

slamora reviewed (changes requested):
What do you think about adding [admin inlines](https://docs.djangoproject.com/en/2.2/ref/contrib/admin/#django.contrib.admin.InlineModelAdmin) to edit word's entries on this screen?

URL: `http://127.0.0.1:8000/admin/linguatec_lexicon/word/4/change/`

![image](https://user-images.githubusercontent.com/4057718/100730192-0843bf00-33ca-11eb-9f3f-aebe9b6c48e8.png)

Right now you should go entry by entry to do that, but I think that common way is to go to a word and edit their related information (entries, diatopic variations...)

AlejandroRG31 committed:
Added inlines methods in admin panel, and changed api and serializer to return word admin url instead of entry

Inlines entries added to word admin panel
Inlines example and verbalconjugation to entry admin panel

AlejandroRG31 commented:
> What do you think about adding [admin inlines](https://docs.djangoproject.com/en/2.2/ref/contrib/admin/#django.contrib.admin.InlineModelAdmin) to edit word's entries on this screen?
> 
> URL: `http://127.0.0.1:8000/admin/linguatec_lexicon/word/4/change/`
> 
> ![image](https://user-images.githubusercontent.com/4057718/100730192-0843bf00-33ca-11eb-9f3f-aebe9b6c48e8.png)
> 
> Right now you should go entry by entry to do that, but I think that common way is to go to a word and edit their related information (entries, diatopic variations...)

Added this method in admin.py in word to show entries and in Entry to show example and verbal conjugations.

Now I understand better why in the other comment you suggested to add a word admin panel url instead of a entry url, definitely with a word admin panel url and its entries deployed with inline method it's so much clear and there is no need to deploy and (edit) link next to every entry, but only next to the word and it allows with inline method to have all the previous info in a more condensed way.

So i changed all previous (edit) urls to just one next to the word term in the frontend
